### PR TITLE
UI Remove obscured json

### PR DESCRIPTION
**Pull request by hashishaw**
### Description
This PR is a follow-on to #28212

It removes obfuscation from the JSON secret detail, since there is now an overview page that prevents users from viewing secret data until they explicitly click the secret tab. 

- [x] ent tests pass

### CODE REVIEW

**Review by Monkeychip — CHANGES_REQUESTED** [review]
Looks like there is still some remaining clean up code. If you search for `allowObscure` you'll see it in the [json-editor.hbs](https://github.com/hashicorp/vault/blob/main/ui/lib/core/addon/components/json-editor.hbs#L28). Also, search for `revealValues`, also in the json-editor (hbs and js). In the json-editor.js you'll see the use of `showObfuscated` which I believe can also be removed. [This was the old PR](https://github.com/hashicorp/vault/pull/25235) that implemented some of the feature—sometimes it can be helpful to use it to track down all the removal.

**Comment by hashishaw** [comment]
> Looks like there is still some remaining clean up code. If you search for `allowObscure` you'll see it in the [json-editor.hbs](https://github.com/hashicorp/vault/blob/main/ui/lib/core/addon/components/json-editor.hbs#L28). Also, search for `revealValues`, also in the json-editor (hbs and js). In the json-editor.js you'll see the use of `showObfuscated` which I believe can also be removed. [This was the old PR](https://github.com/hashicorp/vault/pull/25235) that implemented some of the feature—sometimes it can be helpful to use it to track down all the removal.

I intentionally left in the jsonEditor functionality in case we wanted it somewhere else, but I can just as easily remove it

**Review by Monkeychip — APPROVED** [review]
Thank you for removing the files. I was thinking about our pattern for keeping vs removing unused code last night, and I settled on removing. The history will always exist in Github, and I learning from the wizard removal, any future re-implementation will likely have to deal with a changed set of context/code and acceptance criteria. Thanks for tackling!

**Comment by cwchristerw** [comment]
@hashishaw I would prefer to keep this feature in Vault CE because #28212 is only available in Vault EE. I have secret that has multi-dimentional array (json object). Obscuring JSON has allowed me to see different levels of keys without revealing secrets. I'm not sure but it would nice to keep obscuring even when #28212 is implemented because if subkey is JSON then you couldn't just view only subkeys of subkeys without revealing values :smile:.

**Comment by hellobontempo** [comment]
Hey @cwchristerw - thank you for this feedback! To clarify this: 

> Obscuring JSON has allowed me to see different levels of keys without revealing secrets. 

This obfuscation was _only_ client-side, the browser still read the secret data. We removed it because it was misleading to users who would see in the Vault logs that the secret values were read, even though they never clicked "reveal." 

If you would just like to review the subkeys of a secret I would recommend using curl ([docs](https://developer.hashicorp.com/vault/api-docs/secret/kv/kv-v2#read-secret-subkeys)) 

```
curl \
    --header "X-Vault-Token: ..." \
    https://127.0.0.1:8200/v1/secret/subkeys/my-secret?version=1

**Comment by karabanov** [comment]
I'm sad that secrets are now visible in the JSON editor, now there is a higher chance that strangers will see them, for example, a time tracking program can take a screenshot at the most inopportune moment

In general, I did not understand why it was necessary to remove the masked JSON - why then in normal mode secrets remain masked, but not in JSON mode?

If a secret has nested keys, then the secret viewer immediately opens as JSON and now all secrets are immediately visible there

**Comment by Monkeychip** [comment]
@karabanov Thank you for posting your feedback. As always with KVv2 we had opposing thoughts about a feature. Some hated the extra click to show the values while others appreciated it. After some consideration it was removed for the following reasons:
1. The secret is still being read and obscuring it confused some folks that the "break-glass" of reading the secret didn't occur until the secret was un-obscured. (Though we still default to obscuring on the non-json view).
2. [Subkey + Patch](https://github.com/hashicorp/vault/pull/28212) was released in the UI in 1.18.x which helps solve some of these problems. _Note it's an enterprise only feature in the UI._ 
3. It was causing some regressions because of the use of CodeMirror, our JSON editor, and how state is managed between Ember data and this third party library. 

**TLDR**: one small nice-to-have improvement received negative feedback and caused us several regressions.

However, I will push your feedback forward as I know some folks are sad to see it removed.
